src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.MaxNumber.cs

```diff
@@ -0,0 +1,173 @@
+
+        /// <summary>Computes the element-wise maximum of the numbers in the specified tensors.</summary>
+        /// <param name="x">The first tensor, represented as a span.</param>
+        /// <param name="y">The second tensor, represented as a scalar.</param>
+        /// <param name="destination">The destination tensor, represented as a span.</param>
+        /// <exception cref="ArgumentException">Destination is too short.</exception>
+        /// <exception cref="ArgumentException"><paramref name="x"/> and <paramref name="destination"/> reference overlapping memory locations and do not begin at the same location.</exception>
+        /// <remarks>
+        /// <para>
+        /// This method effectively computes <c><paramref name="destination" />[i] = <typeparamref name="T"/>.MaxNumber(<paramref name="x" />[i], <paramref name="y" />)</c>.
+        /// </para>
+        /// <para>
+        /// The determination of the maximum element matches the IEEE 754:2019 `maximumNumber` function. If either value is <see cref="IFloatingPointIeee754{TSelf}.NaN"/>
+        /// the other is returned. Positive 0 is considered greater than negative 0.
+        /// </para>
+        /// <para>
+        /// This method may call into the underlying C runtime or employ instructions specific to the current architecture. Exact results may differ between different
+        /// operating systems or architectures.
+        /// </para>
+        /// </remarks>
+        public static void MaxNumber<T>(ReadOnlySpan<T> x, T y, Span<T> destination)
+            where T : INumber<T> =>
+            InvokeSpanScalarIntoSpan<T, MaxNumberOperator<T>>(x, y, destination);
+
+        /// <summary>T.MaxNumber(x, y)</summary>
+        internal readonly struct MaxNumberOperator<T> : IAggregationOperator<T> where T : INumber<T>
+        {
+            public static bool Vectorizable => true;
+
+            public static T Invoke(T x, T y) => T.MaxNumber(x, y);
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static Vector128<T> Invoke(Vector128<T> x, Vector128<T> y)
+            {
+                if (AdvSimd.IsSupported)
+                {
+                    if (typeof(T) == typeof(byte)) return AdvSimd.Max(x.AsByte(), y.AsByte()).As<byte, T>();
+                    if (typeof(T) == typeof(sbyte)) return AdvSimd.Max(x.AsSByte(), y.AsSByte()).As<sbyte, T>();
+                    if (typeof(T) == typeof(short)) return AdvSimd.Max(x.AsInt16(), y.AsInt16()).As<short, T>();
+                    if (typeof(T) == typeof(ushort)) return AdvSimd.Max(x.AsUInt16(), y.AsUInt16()).As<ushort, T>();
+                    if (typeof(T) == typeof(int)) return AdvSimd.Max(x.AsInt32(), y.AsInt32()).As<int, T>();
+                    if (typeof(T) == typeof(uint)) return AdvSimd.Max(x.AsUInt32(), y.AsUInt32()).As<uint, T>();
+
+                    if (typeof(T) == typeof(float)) return AdvSimd.MaxNumber(x.AsSingle(), y.AsSingle()).As<float, T>();
```

Comment:
It'd be nice if we could simplify this a bit...

We need to special case `float`/`double`, but all the others can just use `Vector128.Max(x, y)` for example.

Reply:
Maybe we could just have:
```csharp
                if (typeof(T) == typeof(float))
                {
                    if (AdvSimd.IsSupported)
                    {
                        return AdvSimd.MaxNumber(x.AsSingle(), y.AsSingle()).As<float, T>();
                    }

                    return
                        Vector128.ConditionalSelect(Vector128.Equals(x, y),
                            Vector128.ConditionalSelect(IsNegative(y.AsSingle()).As<float, T>(), x, y),
                            Vector128.ConditionalSelect(Vector128.Equals(y.AsSingle(), y.AsSingle()).As<float, T>(), Vector128.Max(x, y), x));
                }

                if (typeof(T) == typeof(double))
                {
                    if (AdvSimd.Arm64.IsSupported)
                    {
                        return AdvSimd.Arm64.MaxNumber(x.AsDouble(), y.AsDouble()).As<double, T>();
                    }

                    return
                        Vector128.ConditionalSelect(Vector128.Equals(x, y),
                            Vector128.ConditionalSelect(IsNegative(y.AsDouble()).As<double, T>(), x, y),
                            Vector128.ConditionalSelect(Vector128.Equals(y.AsDouble(), y.AsDouble()).As<double, T>(), Vector128.Max(x, y), x));
                }

                return Vector128.Max(x, y);
```

Reply:
That's better, thanks.